packages/react-composites/src/composites/ChatComposite/FileDownloadCards.tsx

```diff
@@ -39,7 +41,23 @@ export interface FileDownloadCards {
 export const FileDownloadCards = (props: FileDownloadCards): JSX.Element => {
   const truncateLength = 15;
   const { userId, message } = props;
+  const [showSpinner, setShowSpinner] = useState(false);
   const fileDownloads: FileMetadata[] = message.metadata ? extractFileMetadata(message.metadata) : [];
+  const fileDownloadHandler = useCallback(
+    async (userId, file) => {
+      if (props.downloadHandler) {
+        setShowSpinner(true);
+        const url = await props.downloadHandler(userId, file);
+        if (url instanceof URL) {
+          window.open(url.toString(), '_blank', 'noopener,noreferrer');
+        }
+        setShowSpinner(false);
```

Comment:
`setShowSpinner(false);` can be called immediately after `await props.downloadHandler(userId, file);`

Reply:
done